CDCgov/prime-simplereport
# 5320/johanna remove loinc code from device type

# FRONTEND PULL REQUEST

## Related Issue

- Resolves #5320 

## Changes Proposed
_Changes in Device Type for the admin interface_
- Device Form now uses the types from the graphql generated file and we no longer need to use the local defined `interface Device`.
- In the form data type `testOrderedLoincCode` was marked as optional to match the graphql endpoint definition but the field is still required during validation so nothing has changed for the user.
- `inferLoincCode` method was removed.
- All instances where the `loincCode` was set as a field for the device were removed across multiple components.
- `getDeviceFromDeviceType` method was trimmed down to just map more directly to the device information retrieve from the server.
- Unit testing was updated for multiple components.
- Device related graphql operations were updated to make the field `SupportedDiseaseTestPerformedInput` mandatory and to remove `loincCode`.

_Test Results_
- `GetFacilityResultsForCsvWithCount` query operation was updated to reflect the latest structure of the deviceType object .

_Device Lookup details_
- The Test performed code area was updated to list multiple testOrdered codes and not only covid.
- graphql operation `getDeviceTypesForLookup` was updated to receive the testOrdered information through the field `supportedDiseaseTestPerformed`.

_GraphQL changes in the backend_

- LoincCode was removed from the Input Types that are being used in the mutation resolvers.
- DeviceType model is set to accept null for loincCode (The removal of the field in the DB model will be done in following ticket).
- DeviceTypeService sets the loincCode field as null until next ticket removes the field in the DB.
- Types in main.graphql `CreateDeviceType` `UpdateDeviceType` `DeviceType` were updated to not return `loincCode` and make `supportedDiseaseTestPerformed` mandatory.
- Unit testing was updated.

## Additional Information

- During the implementation of this ticket I hit a weird point where local frontend unit testing were running fine in local but failing in the pipeline or vice versa. As part of the changes you will see small fixes to unit testing of components not really related to the implementation of this ticket.

## Testing
_Deployed in dev7_

- Add a device
- Edit a device
- Add the device to a facility
- Perform a test with the device
- Check the result looks good
- Check the device with the lookup tool

## Screenshots / Demos
Only UI change noticeable to the user is in the device lookup tool

<img width="785" alt="Screenshot 2023-03-20 at 12 55 14 PM" src="https://user-images.githubusercontent.com/103958711/226724636-ba80d1d2-8160-485a-9aa3-4b2e7b15ef1f.png">
